### PR TITLE
1391 ui update allow users to see underneath labels

### PR DESCRIPTION
**Pull request by paarig**
Resolves [#1391](https://github.com/ProjectSidewalk/SidewalkWebpage/issues/1391)

**Changes**

* A button that says "Hide Label" to the left of the zoom buttons
* The button should be greyed (or something similar) when the label is hidden.
* We need the "h" key to to both hide and unhide the label
* The button clicks and keyboard shortcuts need to be logged appropriately, similar to the logging of zoom in/out.

**Testing Instructions**

Test that the changes all work as designed. Specifically, focus on the keyboard shortcuts, the hiding of labels, the proper button coloring, the fact that hide label works properly and independently from validation task to validation task, and logging.

### CODE REVIEW

File: app/views/audit.scala.html [hunk 1]
```diff
@@ -233,7 +233,7 @@ <h4 class="survey-question">@surveyQuestion.surveyQuestionText</h4>
                             <span class="modeSwitch_Name"><span class="underline">N</span>o Sidewalk</span>
                             <div class="ribbon-menu-mode-switch-horizontal-line" val="NoSidewalk"></div>
                         </span>
-                        <span id="ModeSwitchButton_Other" class="modeSwitch" val="Other">
+                        <span id="fModeSwitchButton_Other" class="modeSwitch" val="Other">
```
Comment (misaugstad) [inline app/views/audit.scala.html]: I'm assuming this was an accident?

File: app/views/validation.scala.html [hunk 2]
```diff
@@ -37,6 +37,12 @@
             </div>
             <div id="upper-menu-holder">
                 <div id="upper-menu-title-bar"></div>
+                <div id="label-control-holder"></div>
+                <div id="label-control-holder">
+                    <button id="label-control-button" class="button label-control-button" title="Press the &quot;H&quot; key" data-toggle="tooltip" data-placement="top">
+                        <img src='@routes.Assets.at("javascripts/SVLabel/img/icons/ZoomIn.svg")' class="label-button-icon" alt="Hide Label">
```
Comment (misaugstad) [inline app/views/validation.scala.html]: Needs something other than the zoom in image

File: public/javascripts/SVLabel/src/SVLabel/Main.js [hunk 3]
```diff
@@ -461,7 +461,7 @@ function Main (params) {
         // MissionDescription DOMs
         svl.ui.statusMessage = {};
         svl.ui.statusMessage.holder = $("#current-status-holder");
-        svl.ui.statusMessage.title = $("#current-status-title");
+        svl.ui.statusMessage.title = $("#current-status-title")
```
Comment (misaugstad) [inline public/javascripts/SVLabel/src/SVLabel/Main.js]: I'm assuming this was another mistake?

File: app/views/validation.scala.html [hunk 4]
```diff
@@ -37,6 +37,12 @@
             </div>
             <div id="upper-menu-holder">
                 <div id="upper-menu-title-bar"></div>
+                <div id="label-control-holder"></div>
```
Comment (misaugstad) [inline app/views/validation.scala.html]: I don't think `label-control-holder` and `label-control-button` are descriptive enough. Maybe something like `label-visibility-control-holder` and `label-visibility-control-button`?

File: public/javascripts/SVValidate/src/keyboard/Keyboard.js [hunk 5]
```diff
@@ -58,6 +58,13 @@ function Keyboard(menuUI) {
                     menuUI.agreeButton.removeClass("validate");
                     menuUI.notSureButton.removeClass("validate");
                     break;
+                // "h" key
+                case 72:
+                    svv.labelControl.hideLabel();
+                    svv.tracker.push("KeyboardShortcut_HideLabel", {
```
Comment (misaugstad) [inline public/javascripts/SVValidate/src/keyboard/Keyboard.js]: If we are unhiding the label, we should be recording "KeyboardShortcut_UnhideLabel"

File: public/javascripts/SVValidate/src/label/LabelControl.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        });
+    }
+
+    labelControlButton.on('click', clickHideLabel);
+
+    self.hideLabel = hideLabel;
+    self.refreshLabel = refreshLabel;
+
+    return this;
+}
```
Comment (misaugstad) [inline public/javascripts/SVValidate/src/label/LabelControl.js]: Can you add a newline at the end of the file?

File: public/javascripts/SVValidate/src/label/LabelControl.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+function LabelControl () {
+    var self = this;
+    var visible = true;
+    var labelControlButton = $("#label-control-button");
+
+    /**
+     * Logs interaction when the hide label button is clicked.
+     */
+    function clickHideLabel () {
+        svv.tracker.push("Click_HideLabel");
```
Comment (misaugstad) [inline public/javascripts/SVValidate/src/label/LabelControl.js]: If we are unhiding the label we should be logging "Click_UnhideLabel"

File: app/views/audit.scala.html (hunk 1 above)
Comment (paarig) [reply app/views/audit.scala.html]: Fixed

File: public/javascripts/SVLabel/src/SVLabel/Main.js (hunk 3 above)
Comment (paarig) [reply public/javascripts/SVLabel/src/SVLabel/Main.js]: Fixed

File: app/views/validation.scala.html (hunk 4 above)
Comment (paarig) [reply app/views/validation.scala.html]: Fixed

File: public/javascripts/SVValidate/src/keyboard/Keyboard.js (hunk 5 above)
Comment (paarig) [reply public/javascripts/SVValidate/src/keyboard/Keyboard.js]: Fixed

File: public/javascripts/SVValidate/src/label/LabelControl.js (hunk 6 above)
Comment (paarig) [reply public/javascripts/SVValidate/src/label/LabelControl.js]: Fixed

File: public/javascripts/SVValidate/src/label/LabelControl.js (hunk 7 above)
Comment (paarig) [reply public/javascripts/SVValidate/src/label/LabelControl.js]: Fixed

File: app/views/validation.scala.html (hunk 2 above)
Comment (paarig) [reply app/views/validation.scala.html]: @misaugstad Sent an image in the GitHub task thread that seems to be good. Will incorporate it into the interface once you give me the go ahead.

File: public/javascripts/SVValidate/css/svv-label-control.css [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+#label-visibility-control-holder {
+    background: transparent;
+    position: absolute;
+    left: 0px;
+    top: 30px;
+    height: 55px;
+    width: 185px;
+}
```
Comment (misaugstad) [inline public/javascripts/SVValidate/css/svv-label-control.css]: Newline at end of this file please!

File: public/javascripts/SVValidate/src/label/LabelControl.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
+/**
+ * Handles the hiding and showing of labels in the Google StreetView panorama.
+ * This is also called by the Keyboard class to deal with hiding the label
+ * via keyboard shortcuts.
+ * @returns {LabelControl}
+ * @constructor
+ */
+function LabelControl () {
```
Comment (misaugstad) [inline public/javascripts/SVValidate/src/label/LabelControl.js]: You changed the names in the html and css from "label control" to "label visibility control", but you didn't do it in the JavaScript yet.

**Comment by misaugstad** [comment]
@paarig for future PRs, when addressing feedback, still try to make your commit messages more descriptive. Using phrases like "improved code style" or "renamed some variables", etc. And if there are multiple distinct changes to be made, try to do them in separate commits as well.

File: public/javascripts/SVValidate/src/label/LabelControl.js (hunk 9 above)
Comment (paarig) [reply public/javascripts/SVValidate/src/label/LabelControl.js]: Fixed

File: public/javascripts/SVValidate/css/svv-label-control.css (hunk 8 above)
Comment (paarig) [reply public/javascripts/SVValidate/css/svv-label-control.css]: Fixed

**Comment by misaugstad** [comment]
Just tested the functionality, and it seems good. Nice work!

**Pull request merged**